src/cmd/linuxkit/moby/build.go

```diff
@@ -427,6 +453,53 @@ func tarAppend(iw *tar.Writer, tr *tar.Reader) error {
 	return nil
 }
 
+// Attempt to decompress a Linux kernel image
+// arm64 kernel images are plain gzip
+// x86 kernel images are in bzImage format, where the offset for the gzip'ed image is described in the header
```

Comment:
FWIW it can actually be one of several possible compression algs. Fine to just support one for now though.

e.g. on Debian:
```
$ grep CONFIG_KERNEL /boot/config-4.17.0-1-amd64
# CONFIG_KERNEL_GZIP is not set
# CONFIG_KERNEL_BZIP2 is not set
# CONFIG_KERNEL_LZMA is not set
CONFIG_KERNEL_XZ=y
# CONFIG_KERNEL_LZO is not set
# CONFIG_KERNEL_LZ4 is not set
```